src/test/scala/io/iohk/ethereum/jsonrpc/JsonRpcControllerFixture.scala

```diff
@@ -0,0 +1,126 @@
+package io.iohk.ethereum.jsonrpc
+
+import akka.actor.ActorSystem
+import akka.testkit.TestProbe
+import akka.util.ByteString
+import io.iohk.ethereum.{Fixtures, Timeouts}
+import io.iohk.ethereum.blockchain.sync.EphemBlockchainTestSetup
+import io.iohk.ethereum.consensus.{ConsensusConfigs, TestConsensus}
+import io.iohk.ethereum.consensus.ethash.blocks.EthashBlockGenerator
+import io.iohk.ethereum.consensus.ethash.validators.ValidatorsExecutor
+import io.iohk.ethereum.crypto.ECDSASignature
+import io.iohk.ethereum.db.storage.AppStateStorage
+import io.iohk.ethereum.domain.{Block, BlockBody, SignedTransaction}
+import io.iohk.ethereum.jsonrpc.JsonRpcController.JsonRpcConfig
+import io.iohk.ethereum.keystore.KeyStore
+import io.iohk.ethereum.ledger.{BloomFilter, Ledger, StxLedger}
+import io.iohk.ethereum.utils.{Config, FilterConfig}
+import org.bouncycastle.util.encoders.Hex
+import org.json4s.JsonAST.{JArray, JInt, JString, JValue}
+import org.scalamock.scalatest.MockFactory
+
+import scala.concurrent.duration._
+
+trait JsonRpcControllerFixture extends MockFactory with EphemBlockchainTestSetup with JsonMethodsImplicits {
+  def config: JsonRpcConfig = JsonRpcConfig(Config.config)
+
+  def rawTrnHex(xs: Seq[SignedTransaction], idx: Int): Option[JString] =
+    xs.lift(idx)
+      .map(encodeSignedTrx)
+
+  def encodeSignedTrx(x: SignedTransaction) =
+    encodeAsHex(RawTransactionCodec.asRawTransaction(x))
+
+  val version = Config.clientVersion
+  val blockGenerator = mock[EthashBlockGenerator]
+
+  override implicit lazy val system = ActorSystem("JsonRpcControllerSpec_System")
```

Comment:
Do we need to create ActorSystem for each test?

Reply:
This.
I'd say it's preferrable to use akka's `TestKit` on suite level and use there `WithActorSystemShutdown` trait.
Benefits are as follows:
1. Creating and shutting down systems is quite expensive operation.
2. Actors of any kind are created per spec anyways so lack of isolation is not an issue I think
3. In other project we had real problems with tests due to leaking running actor systems

Reply:
How about merging this PR and fix the unnecessary spawning actor system in a separate one.

There is a pretty complex wiring using cake pattern (actor system that comes from `ActorSystemBuilder` (via `StdTestConsensusBuilder` via `ScenarioSetup` via `EphemBlockchainTestSetup` and ActorSystemBilder is required in `StdLedgerBuilder` via self type). I need some time to figure out how to do this.

Reply:
Sounds good to me. And such task/PR could be extended to making sure actor system is properly closed in all tests.

Reply:
PR for this change: https://github.com/input-output-hk/mantis/pull/751